src/v/coproc/event_listener.cc

```diff
@@ -196,11 +202,40 @@ ss::future<> event_listener::do_start() {
         } else {
             _active_ids.clear();
             _offset = model::offset(0);
+            co_await advertise_state_update();
         }
     }
+
     co_await do_ingest();
 }
 
+ss::future<> event_listener::boostrap_status_topic() {
+    /// Bootstrap the coproc status topic. At mininum the wasm engine must
+    /// be up for this part of code to be reached.
+    kafka::creatable_topic copro_status_topic{
+      .name = model::coprocessor_status_topic,
+      .num_partitions = 1,
+      .replication_factor = 3,
+      .configs{
+        kafka::createable_topic_config{
+          .name = "cleanup.policy", .value = "compact"},
+        kafka::createable_topic_config{
+          .name = "compression.type", .value = "zstd"}}};
```

Comment:
I'm not too sure that this compression setting has any affect. You may want to verify this and then possibly create some tickets to add support where necessary (e.g. leave it as default producer value, and then add compression to our client, or figure out what topic-level compression settings are supposed to do).

Reply:
This is what `rpk wasm deploy` does when bootstrapping the `coprocessor_internal_topic`.

Reply:
then are both cases not actually doing anything?

Reply:
How does compression work? At the client end or broker end?

Reply:
Quick google says both are possible in kafka

Reply:
does it do anything in redpanda?

Reply:
I think I understand now after grepping the source. compression.type doesn't have an effect, but the cleanup.policy will do what I intend to do. I just blindly copiied these settngs from some wasm go code i think i must update that as well. https://github.com/graphcareful/redpanda/blob/dev/src/go/rpk/pkg/cli/cmd/wasm/common.go#L33